### PR TITLE
 🐛 preserve file names from server forwards

### PR DESCRIPTION
**Pull request by fiendish**
variant solution for #184

### CODE REVIEW

**Comment by fiendish** [comment]
I can try but your tests will fail because you test file extensions.

**Comment by fiendish** [comment]
ugh. the merge was bad lol. fixing...

**Comment by fiendish** [comment]
ok I did it

File: kf_lib_data_ingest/common/file_retriever.py [hunk 1]
```diff
@@ -84,52 +84,50 @@ def _s3_save(protocol, source_loc, dest_obj, auth=None, logger=None):
...
+            filename = cdisp_params.get('filename*')
+            # RFC 5987 ext-parameter is actually more complicated than this,
+            # but this should get us 90% there, and the rfc6266 python lib is
+            # broken after PEP 479. *sigh* - Avi K
+            if filename and filename.lower.startswith('utf-8'):
+                filename = filename.split("'", 2)[2].decode('utf-8')
+            else:
+                filename = cdisp_params.get('filename')
 
             # Header did not provide filename
```
Comment (znatty22) [inline kf_lib_data_ingest/common/file_retriever.py]: Remove or move comment to else block

File: kf_lib_data_ingest/common/file_retriever.py [hunk 2]
```diff
@@ -84,52 +84,50 @@ def _s3_save(protocol, source_loc, dest_obj, auth=None, logger=None):
...
-            if len(parts) < 2:
+            if filename:
+                dest_obj.original_name = filename
```
Comment (znatty22) [inline kf_lib_data_ingest/common/file_retriever.py]: To be consistent with what gets assigned to original name using other getters:
```suggestion
                dest_obj.original_name = source_loc
```

**Comment by znatty22** [comment]
@fiendish if you agree with the requested changes, I can just merge this into my branch and make the changes there so you don't have to

**Pull request merged**